subtests/docker_cli/run_cgroups/cgroup_parent.py

```diff
@@ -0,0 +1,103 @@
+"""
+Tests for bz1320302 - odd behavior of --cgroup-parent option
+
+Operational Summary
+----------------------
+
+#. Start container with --cgroup-parent=X, for various values of X
+#. run container command 'tail -n1 /proc/1/cgroup'
+#. Compare to expected result
+"""
+
+
+import os.path
+import re
+from autotest.client import utils
+from dockertest.dockercmd import DockerCmd
+from dockertest.images import DockerImage
+from dockertest.output import OutputGood
+from dockertest.xceptions import DockerTestError, DockerTestFail
+from cgroups_base import cgroups_base
+
+
+class cgroup_parent_base(cgroups_base):
+    def _setup(self, cg_parent, expected_path, expected_stderr):
+        """
+        Set up test parameters. Our caller gives us a value to use
+        for --cgroup-parent, and what to expect in stdout & stderr.
+        We replace the format strings {rand1} and {rand2} with two
+        pseudorandom strings, and {cid} with the container ID (once
+        we have it).
+        """
+        self.sub_stuff['rand1'] = utils.generate_random_string(8)
+        self.sub_stuff['rand2'] = utils.generate_random_string(8)
+        self.sub_stuff['cgroup_parent'] = cg_parent.format(**self.sub_stuff)
+        self.sub_stuff['expected_path'] = expected_path
+        self.sub_stuff['expected_stderr'] = expected_stderr
+
+    def run_once(self):
+        """
+        Run docker with the given --cgroup-parent; preserve cgroup info.
+        """
+        super(cgroup_parent_base, self).run_once()
+
+        cidfile = os.path.join(self.tmpdir, 'cidfile')
+        self.sub_stuff['cidfile'] = cidfile
+
+        subargs = ['--rm',
+                   '--cgroup-parent=%s' % self.sub_stuff['cgroup_parent'],
+                   '--cidfile', cidfile,
+                   DockerImage.full_name_from_defaults(self.config),
+                   '/bin/cat', '/proc/1/cgroup']
+        dc = DockerCmd(self, "run", subargs,
+                       timeout=self.config['docker_timeout'])
+        self.sub_stuff["cmdresult"] = dc.execute()
+
+    def postprocess(self):
+        super(cgroup_parent_base, self).postprocess()
+
+        cmdresult = self.sub_stuff["cmdresult"]
+        OutputGood(cmdresult, ignore_error=False)
+
+        # Expected output may include the container ID; allow caller
+        # to use "{cid}", which we replace with actual one for this run.
+        cid = open(self.sub_stuff['cidfile'], 'rb').read().strip()
+        if len(cid) < 12:
+            raise DockerTestError("bad cid (length < 12) in --cidfile")
+        self.sub_stuff['cid'] = cid
+        path_exp = self.sub_stuff['expected_path'].format(**self.sub_stuff)
+        stderr_exp = self.sub_stuff['expected_stderr'].format(**self.sub_stuff)
```

Comment:
This will make expanding/extending the test later (if necessary) a lot easier to do.